src/azure.eventhub/client.bal

```diff
@@ -0,0 +1,378 @@
+        req.addHeader("Authorization", token);
+        var response = clientEndpoint->get("/", req);
+        if (response is http:Response) {
+            var xmlPayload = response.getXmlPayload();
+            if (xmlPayload is xml) {
+                return xmlPayload;
+            }
+            return error(EVENT_HUB_ERROR, message = "invalid response from EventHub API. status code: " + response.statusCode.toString()
+                + ", payload: " + response.getTextPayload().toString());
+        } else {
+            return error(EVENT_HUB_ERROR, message = "error invoking EventHub API ", cause = response);
+        }
+    }
+
+    # Get details of revoked publisher
+    #
+    # + return - Return revoke publisher or Error
+    public remote function getRevokedPublishers() returns @tainted xml|error {
+        http:Client clientEndpoint = new ("https://" + self.config.resourceUri);
+        http:Request req = new;
+        string token = self.getSASToken();
+        req.addHeader("Authorization", token);
+        var response = clientEndpoint->get("/revokedpublishers", req);
+        if (response is http:Response) {
+            var xmlPayload = response.getXmlPayload();
+            if (xmlPayload is xml) {
+                return xmlPayload;
+            }
+            return error(EVENT_HUB_ERROR, message = "invalid response while getting revoked publishers: status code: " + response.statusCode.toString()
+                + ", payload: " + response.getTextPayload().toString());
+        } else {
+            return error(EVENT_HUB_ERROR, message = "error invoking EventHub API ", cause = response);
+        }
+    }
+
+    # Revoke a publisher
+    #
+    # + publisherName - publisher name 
+    # + return - Return revoke publisher details or error
+    public remote function revokePublisher(string publisherName) returns @tainted xml|error {
+        http:Client clientEndpoint = new ("https://" + self.config.resourceUri);
+        http:Request req = new;
+        string token = self.getSASToken();
+        req.addHeader("Authorization", token);
+        var response = clientEndpoint->put("/revokedpublishers/" + publisherName, req);
+        if (response is http:Response) {
+            var xmlPayload = response.getXmlPayload();
+            if (xmlPayload is xml) {
+                return xmlPayload;
+            }
+            return error(EVENT_HUB_ERROR, message = "invalid response while revoking publisher: " + publisherName
+                + ". " + response.statusCode.toString());
+        } else {
+            return error(EVENT_HUB_ERROR, message = "error invoking EventHub API ", cause = response);
+        }
+    }
+
+    # Resume a publisher
+    #
+    # + publisherName - publisher name 
+    # + return - Return publisher details or error
+    public remote function resumePublisher(string publisherName) returns @tainted xml|error {
+        http:Client clientEndpoint = new ("https://" + self.config.resourceUri);
```

Comment:
Can we keep the client at the object body level than creating it in every remote method ?